test/integration/simple_operations_test.lua

```diff
@@ -241,136 +259,162 @@ add('test_update', function(g)
+    local result, err = g.cluster.main_server.net_box:eval([[
         local crud = require('crud')
         return crud.replace('customers', {id = 44, name = 'Jane Doe', age = 18})
     ]])
 
     t.assert_equals(err, nil)
-    t.assert_covers(obj, {id = 44, name = 'Jane Doe', age = 18})
-    t.assert(type(obj.bucket_id) == 'number')
+    local objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {{id = 44, name = 'Jane Doe', age = 18, bucket_id = 2805}})
+    t.assert(type(objects[1].bucket_id) == 'number')
 end)
 
 add('test_upsert', function(g)
     -- upsert tuple not exist
-    local obj, err = g.cluster.main_server.net_box:eval([[
+    local result, err = g.cluster.main_server.net_box:eval([[
         local crud = require('crud')
         return crud.upsert('customers', {id = 66, name = 'Jack Sparrow', age = 25}, {
             {'+', 'age', 25},
             {'=', 'name', 'Leo Tolstoy'}
         })
     ]])
 
-    t.assert_equals(obj, nil)
+    t.assert_equals(#result.rows, 0)
+    t.assert_equals(result.metadata, {
+        {name = 'id', type = 'unsigned'},
+        {name = 'bucket_id', type = 'unsigned'},
+        {name = 'name', type = 'string'},
+        {name = 'age', type = 'number'},
+    })
     t.assert_equals(err, nil)
 
     -- get
-    local obj, err = g.cluster.main_server.net_box:eval([[
+    local result, err = g.cluster.main_server.net_box:eval([[
         local crud = require('crud')
         return crud.get('customers', 66)
     ]])
     t.assert_equals(err, nil)
-    t.assert_covers(obj, {id = 66, name = 'Jack Sparrow', age = 25})
-    t.assert(type(obj.bucket_id) == 'number')
+    local objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {{id = 66, name = 'Jack Sparrow', age = 25, bucket_id = 486}})
+    t.assert(type(objects[1].bucket_id) == 'number')
 
     -- upsert same query second time tuple exist
-    local obj, err = g.cluster.main_server.net_box:eval([[
+    local result, err = g.cluster.main_server.net_box:eval([[
         local crud = require('crud')
         return crud.upsert('customers', {id = 66, name = 'Jack Sparrow', age = 25}, {
             {'+', 'age', 25},
             {'=', 'name', 'Leo Tolstoy'}
         })
     ]])
 
-    t.assert_equals(obj, nil)
+    t.assert_equals(#result.rows, 0)
     t.assert_equals(err, nil)
 
     -- get
-    local obj, err = g.cluster.main_server.net_box:eval([[
+    local result, err = g.cluster.main_server.net_box:eval([[
         local crud = require('crud')
         return crud.get('customers', 66)
     ]])
     t.assert_equals(err, nil)
-    t.assert_covers(obj, {id = 66, name = 'Leo Tolstoy', age = 50})
-    t.assert(type(obj.bucket_id) == 'number')
+    local objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {{id = 66, name = 'Leo Tolstoy', age = 50, bucket_id = 486}})
+    t.assert(type(objects[1].bucket_id) == 'number')
```

Comment:
You've just checked that `bucket_id` us 487.

Reply:
Removed